src/config/configProcessors.js

```diff
@@ -0,0 +1,133 @@
+          LogLevel.ERROR);
+    }
+    constructor = DEFAULT_STORAGE_INTERFACES[constructor];
+  }
+  try {
+    return new constructor(params);
+  } catch (err) {
+    log(`Could not construct a new instance of the class ` +
+        `${constructor} with parameter`, LogLevel.ERROR);
+    log(params, LogLevel.ERROR);
+    log(err, LogLevel.ERROR);
+    return null;
+  }
+}
+
+
+/**
+ * Construct an event processor from a representative string or constructor,
+ * with the given parameters.
+ *
+ * @private
+ * @param {!ProcessorConstructor|string} constructor
+ * @param {!Object<string, *>} params
+ * @return {?EventProcessor}
+ */
+function buildProcessor_(constructor, params) {
+  if (typeof constructor === 'string') {
+    if (!(constructor in DEFAULT_EVENT_PROCESSORS)) {
+      log(`No event processor with name ${constructor} found.`,
+          LogLevel.ERROR);
+    }
+    constructor = DEFAULT_EVENT_PROCESSORS[constructor];
+  }
+  try {
+    return new constructor(params);
+  } catch (err) {
+    log(`Could not construct a new instance of the class ` +
+        `${constructor} with parameter`, LogLevel.ERROR);
+    log(params, LogLevel.ERROR);
+    log(err, LogLevel.ERROR);
+    return null;
+  }
+}
+
+/**
+ * When called with an implementation of the eventProcessor and
+ * storageInterface API, this function registers processors to react to the
+ * 'set' and 'event' commands.
+ *
+ * @param {!DataLayerHelper} helper The data layer helper object we reference
+ * @param {!StorageConstructor|string} eventProcessor The event processor to
+ *     use when responding to event commands and deciding how keys should be
+ *     stored, or a string representing an object of this type.
+ * @param {!Object<string, *>} eventOptions Options objects to pass to the
+ *    event processor.
+ * @param {!StorageConstructor|string} storageInterface An interface to a class
+ *     that will handle setting and loading long term data, or a string
+ *     representing an object of this type.
+ * @param {!Object<string, *>} storageOptions Options objects to pass to the
+ *    storage interface.
+ */
+function configProcessors(helper, eventProcessor, eventOptions,
+                          storageInterface, storageOptions) {
+  // TODO: Read in data from the model as extra options
+  const processor = buildProcessor_(eventProcessor, eventOptions);
+  const storage = buildStorage_(storageInterface, storageOptions);
+  if (!processor || !storage) {
+    // A failure occurred, return now to prevent the page from crashing.
+    return;
+  }
+
+  /**
+   * Process an event object by sending it to the registered event processor
+   * along with interfaces to the storage and data layer helper.
+   *
+   * @param {string} name The name of the event to process.
+   * @param {!Object<string, *>=} options The options object describing
+   *     the event.
+   */
+  function processEvent(name, options = undefined) {
```

Comment:
I think a JSDoc in the middle of a function decreases readability (it looks like a class method) and doesn't provide a whole lot of benefit. I know this function is small, but if you think it needs a JSDoc, please take it out and make it a private module-level function. Otherwise, a normal comment is fine.

Also, for nested functions, please use an arrow function.

Reply:
This function uses this and cannot be made an arrow function. I removed the jsdoc.